text/0089-metrics-views-api.md

```diff
@@ -0,0 +1,264 @@
+
+We require aggregations to be mergeable so that they produce the same results regardless of the collection interval, or the number of collection events per export interval.
+
+### Aggregating across label keys
+
+Every measurement is associated with a _LabelSet_, a set of key-value pairs that describes the environment in which the measurement was captured.
+Label keys and values may be extracted from the [correlation context](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-correlationcontext.md) at the time of measurement.
+They may also be set by the user [at the time of measurement](https://github.com/open-telemetry/opentelemetry-specification/blob/f4aeb318a5b77c9c39132a8cbc5d995e222d124f/specification/api-metrics-user.md#direct-instrument-calling-convention) or at the time at which [the metric instrument is bound](https://github.com/open-telemetry/opentelemetry-specification/blob/f4aeb318a5b77c9c39132a8cbc5d995e222d124f/specification/api-metrics-user.md#bound-instrument-calling-convention).
+
+For each registered view, in each collection interval, measurements with the same labelset are aggregated together.
+
+The view API should allow the user to specify the set of label keys to track for a given view.
+Other label keys should be dropped before aggregation.
+By default, if the user doesn't specify a set of label keys to track, the aggregator should record all label keys.
+
+Users have three options to configure a view's label keys:
+
+1. Record all label keys.
+    This is the default option, and the behavior of the "ungrouped" integrator in [opentelemetry-specification#347](https://github.com/open-telemetry/opentelemetry-specification/pull/347).
+2. Specify a set of label keys to track at view creation time, and drop other keys from the labelsets of recorded measurements before aggregating.
+    This is the behavior of the "defaultkeys" integrator in [opentelemetry-specification#347](https://github.com/open-telemetry/opentelemetry-specification/pull/347).
+3. Drop all label keys, and aggregate all measurements from the metric instrument together regardless of their labelsets.
+    This is equivalent to using an empty list with option 2.
+
+Because we don't require the user to specify the set of label keys up front, and because we don't prevent users from recording measurements with missing labels in the API, some label values may be undefined.
+Aggregators should preserve undefined label values, and exporters may convert them as required by the backend.
+
+For example, consider a Sum-aggregated Counter instrument that captures four consecutive measurements:
+
+1. `{labelset: {'k1': 'v11'}, value: 1}`
+2. `{labelset: {'k1': 'v12'}, value: 10}`
+3. `{labelset: {'k1': 'v11', 'k2': 'v21'}, value: 100}`
+4. `{labelset: {'k1': 'v12', 'k2': 'v22'}, value: 1000}`
+
+And consider three different views, one that tracks label key _k1_, one that tracks label key _k2_, and one that tracks both.
+
```

Comment:
From this example, can I assume that the labelset reduction/drop **should** occur at the Aggregator itself. 
i.e If my only view is add(k1), I'll end up with just 2 time series - as there were 2 unique values for k1.

OR, does the aggregator still keep all dimensions (storing k1_uniquecount * k2_uniquecount timeseries), and exporter/batcher drops the dimensions?